test/pkg/environment/common/expectations.go

```diff
@@ -565,34 +565,73 @@ func (env *Environment) ConsistentlyExpectNodeCount(comparator string, count int
 	return lo.ToSlicePtr(nodeList.Items)
 }
 
-func (env *Environment) ConsistentlyExpectNoDisruptions(nodeCount int, duration time.Duration) (taintedNodes []*corev1.Node) {
+// ConsistentlyExpectNoDisruptions asserts that the number of tainted nodes remains the same.
+// And that the number of nodeclaims remains the same.
+func (env *Environment) ConsistentlyExpectNoDisruptions(nodeCount int, duration time.Duration) {
 	GinkgoHelper()
-	return env.ConsistentlyExpectDisruptionsWithNodeCount(0, nodeCount, duration)
+	Consistently(func(g Gomega) {
+		nodeClaimList := &karpv1.NodeClaimList{}
+		g.Expect(env.Client.List(env, nodeClaimList, client.HasLabels{test.DiscoveryLabel})).To(Succeed())
+		g.Expect(nodeClaimList.Items).To(HaveLen(nodeCount))
+		nodeList := &corev1.NodeList{}
+		g.Expect(env.Client.List(env, nodeList, client.HasLabels{test.DiscoveryLabel})).To(Succeed())
+		g.Expect(nodeList.Items).To(HaveLen(nodeCount))
+		nodeList.Items = lo.Filter(nodeList.Items, func(n corev1.Node, _ int) bool {
+			_, ok := lo.Find(n.Spec.Taints, func(t corev1.Taint) bool {
+				return t.MatchTaint(&karpv1.DisruptedNoScheduleTaint)
+			})
+			return ok
+		})
+		g.Expect(nodeList.Items).To(HaveLen(0))
+	}, duration).Should(Succeed())
 }
 
-// ConsistentlyExpectDisruptionsWithNodeCount will continually ensure that there are exactly disruptingNodes with totalNodes (including replacements and existing nodes)
-func (env *Environment) ConsistentlyExpectDisruptionsWithNodeCount(disruptingNodes, totalNodes int, duration time.Duration) (taintedNodes []*corev1.Node) {
+// ConsistentlyExpectDisruptionsWithNodeCount consistently ensures a max on number of concurrently disrupting and non-terminating nodes.
+// For example: if we have 5 nodes, with a budget of 2 nodes, we ensure that `disruptingNodes <= 2`
+// We use nonDisrupting to assert that we're not creating too many instances in replacement.
+func (env *Environment) ConsistentlyExpectDisruptionsUntilTarget(disruptingNodes, startingPoint, target int, timeout time.Duration) {
```

Comment:
This says consistently, but then doesn't do a consistently check/does an eventually so do we need to change this function name?

Reply:
Got it -- I think I might understand why you used this phrasing because you are continually doing a check up to a point but also bailing if you don't satisfy a different condition